lib/livebook_web/live/settings_live/env_var_component.ex

```diff
@@ -0,0 +1,92 @@
+defmodule LivebookWeb.SettingsLive.EnvVarComponent do
+  use LivebookWeb, :live_component
+
+  alias Livebook.Settings
+  alias Livebook.Settings.EnvVar
+
+  @impl true
+  def update(assigns, socket) do
+    {env_var, operation} =
+      if assigns.env_var,
+        do: {assigns.env_var, :edit},
+        else: {%EnvVar{}, :new}
+
+    changeset = Settings.change_env_var(env_var)
+
+    {:ok,
+     socket
+     |> assign(assigns)
+     |> assign(changeset: changeset, env_var: env_var, operation: operation)}
+  end
+
+  @impl true
+  def render(assigns) do
+    ~H"""
+    <div class="p-6 flex flex-col space-y-5">
+      <h3 class="text-2xl font-semibold text-gray-800">
+        <%= if @operation == :new, do: "Add environment variable", else: "Edit environment variable" %>
+      </h3>
+      <p class="text-gray-700">
+        Configure your application global environment variables.
+      </p>
+      <.form
+        id={"#{@id}-form"}
+        let={f}
+        for={@changeset}
+        phx-target={@myself}
+        phx-submit="save"
+        phx-change="validate"
+        autocomplete="off"
+        spellcheck="false"
+      >
+        <div class="flex flex-col space-y-4">
+          <.input_wrapper form={f} field={:key} class="flex flex-col space-y-1">
+            <div class="input-label">
+              Key <span class="text-xs text-gray-500">(alphanumeric and underscore)</span>
+            </div>
+            <%= text_input(f, :key, class: "input") %>
```

Comment:
it would be nice if this field gets the focus when opening up the modal so we can just start typing.

Reply:
What if we add focus to `:key` when is to add a new env var and focus `:value` when is to edit an env var?

Reply:
Sound great